avm/res/key-vault/vault/access-policy/main.bicep

```diff
@@ -35,3 +28,31 @@ output name string = policies.name
 
 @description('The resource ID of the access policies assignment.')
 output resourceId string = policies.id
+
+// ================ //
+// Definitions      //
+// ================ //
+type accessPoliciesType = {
+  @description('Required. The tenant ID that is used for authenticating requests to the key vault.')
+  tenantId: string
```

Comment:
Hey @johnlokerse, thanks for the contribution. You're right that the UDT takes care of the formattedAccessPolicies with one exception (which may have been the original reason it was there in the first place), that is, the default for the tenantId `tenant().tenantId`.

It would be nice from a usage perspective if we could still keep this feature. However, this would require us to keep the loop.

Reply:
Ah yes, I get it. I will revert it back!

Reply:
I might have an idea. Will test some cases and report back later :-)

Reply:
I have reverted the code back. Had an idea of using the map function, and checking if `tenantId` exists but realized I was writing the same code but in a different way and it went way more complex than a simple for-loop. Also tried something with the string `replace` function.

All in all, a `tenantId` property and value check and a for-loop are needed and I think the current solution is the way to go.

I have replaced the `contains` for the safe-dereference operator.